File: 0027/0027.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2020/12/28 17:47
# File: 0027
# Desc: 

class Solution:
    def removeElement(self, nums: list, val: int) -> int:
        i = 0
        num = 0
        while i < len(nums):
            if not val == nums[i]:
                num += 1
                i += 1
            else:
                nums.remove(nums[i])
        return num

File: Solutions/0027/0027.java
// Author: washing
// DateTime: 2021/4/19 10:10
// File: 0027.java
// Desc: 

class Solution {
    public int removeElement(int[] nums, int val) {
        int r = nums.length-1;
        int l = 0;
        while (l <= r){
            if (nums[r] == val) r--;
            else {
                if (nums[l] == val){
                    nums[l] = nums[r];
                    r--;
                }
                l++;
            }
        }
        return l;
    }
}